src/Tests/dotnet-publish.Tests/GivenDotnetPublishPublishesProjects.cs

```diff
@@ -137,6 +140,83 @@ public void ItPublishesSelfContainedWithRid(string args)
                      .And.HaveStdOutContaining("Hello World");
         }
 
+        [Fact]
+        public void ItPublishesSelfContainedWithPublishSelfContainedTrue()
+        {
+            var testAppName = "MSBuildTestApp";
+            var rid = EnvironmentInfo.GetCompatibleRid();
+            var outputDirectory = PublishApp(testAppName, rid, "-p:PublishSelfContained=true");
+
+            var outputProgram = Path.Combine(outputDirectory.FullName, $"{testAppName}{Constants.ExeSuffix}");
+
+            outputDirectory.Should().HaveFiles(new[] {
+                "System.dll", // File that should only exist if self contained 
+            });
+
+            new RunExeCommand(Log, outputProgram)
+                .Execute()
+                .Should().Pass()
+                     .And.HaveStdOutContaining("Hello World");
+        }
+
+        [Theory]
+        [InlineData("net7.0", true, false, false)]
+        [InlineData("net7.0", false, false, false)]
+        [InlineData("net7.0", true, true, false)]
+        [InlineData("net7.0", true, true, true)]
+        public void PublishSelfContainedPropertyDoesOrDoesntOverrideSelfContainProperty(string targetFramework, bool publishSelfContained, bool selfContainedIsGlobal, bool publishSelfContainedIsGlobal)
+        {
+            var rid = EnvironmentInfo.GetCompatibleRid(targetFramework);
+
+            var testAsset = _testAssetsManager
+            .CopyTestAsset("HelloWorld", identifier: $"PSC-OVERRIDES-{publishSelfContained}-{selfContainedIsGlobal}-{publishSelfContainedIsGlobal}")
+            .WithSource()
+            .WithProjectChanges(project =>
+            {
+                var ns = project.Root.Name.Namespace;
+                var propertyGroup = project.Root.Elements(ns + "PropertyGroup").First();
+                propertyGroup.Add(new XElement(ns + "SelfContained", (!publishSelfContained).ToString()));
+                propertyGroup.Add(new XElement(ns + "PublishSelfContained", publishSelfContained.ToString()));
+            });
+
+            var publishCommand = new PublishCommand(testAsset);
+            List<string> args = new List<string>
+            {
+                $"/p:RuntimeIdentifier={rid}",
+                $"/p:_IsPublishing=true", // Normally this would be set by the CLI (OR VS Soon TM), but this calls directly into t:/Publish.
```

Comment:
Instead of using PublishCommand, I would probably use `dotnet publish` here so that you don't have to explicitly add this property.  You can do something like

```c#
new DotnetCommand(Log, "publish")
  .WithWorkingDirectory(testAsset.Path)
  .Execute(args)
  .Should()
  .Pass();
```